connectd/connectd.c

```diff
@@ -976,8 +983,43 @@ static void try_connect_one_addr(struct connecting *connect)
 			af = AF_INET6;
 			break;
 		case ADDR_TYPE_DNS:
-			// TODO: resolve with getaddrinfo and set af
-			break;
+			/* Resolve with getaddrinfo */
+			memset(&hints, 0, sizeof(hints));
```

Comment:
This misses two import cases:

always-use-proxy -> don't resolve, just hand it to the proxy.
!use->dns -> skip this entry.

Reply:
good catch... will fix

Reply:
Do we want testcase that checks DNS is really skipped or passed to proxy if configured or is my recent change sufficient?

Reply:
Hm, yes I need a testcase because the `if (!use_dns) goto next;` does not work as `bool use_dns = connect->daemon->use_dns;`  seems to always false.

Will fix that than also add a test...

Reply:
So I investiaged a bit:
`disable-dns` is set by pytest utils `contrib/pyln-testing/pyln/testing/utils.py` line 46. This makes the testcase fail now, as pytest tells any lightningd it spawns not to use DNS at all.

So the code is right, but the test can't work.
What should I do about it?

Second minor issue I saw is that the connect command (for a DNS only node) fails correctly but with an empty error message.

Reply:
@rustyrussell
I fixed that and added a working testcase for DNS enabled and disabled.